src/main/resources/archetype-resources/__rootArtifactId__/src/main/java/__pluginFormat__/internal/proxy/__pluginClass__ProxyFacetImpl.java

```diff
@@ -0,0 +1,223 @@
+import org.sonatype.nexus.repository.cache.CacheInfo;
+import org.sonatype.nexus.repository.config.Configuration;
+import org.sonatype.nexus.repository.proxy.ProxyFacet;
+import org.sonatype.nexus.repository.proxy.ProxyFacetSupport;
+import org.sonatype.nexus.repository.storage.Asset;
+import org.sonatype.nexus.repository.storage.Bucket;
+import org.sonatype.nexus.repository.storage.Component;
+import org.sonatype.nexus.repository.storage.StorageFacet;
+import org.sonatype.nexus.repository.storage.StorageTx;
+import org.sonatype.nexus.repository.storage.TempBlob;
+import org.sonatype.nexus.repository.transaction.TransactionalStoreBlob;
+import org.sonatype.nexus.repository.transaction.TransactionalTouchBlob;
+import org.sonatype.nexus.repository.transaction.TransactionalTouchMetadata;
+import org.sonatype.nexus.repository.view.Content;
+import org.sonatype.nexus.repository.view.Context;
+import org.sonatype.nexus.repository.view.matchers.token.TokenMatcher;
+import org.sonatype.nexus.transaction.UnitOfWork;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.sonatype.nexus.plugins.${pluginFormat}.internal.util.${pluginClass}PathUtils.*;
+import static org.sonatype.nexus.repository.storage.AssetEntityAdapter.P_ASSET_KIND;
+
+/**
+ * ${pluginClass} {@link ProxyFacet} implementation.
+ *
+ * @since 0.0.1
+ */
+@Named
+public class ${pluginClass}ProxyFacetImpl
+    extends ProxyFacetSupport
+    implements ${pluginClass}ProxyFacet
+{
+  private ${pluginClass}PathUtils ${pluginFormat}PathUtils;
+
+  private ${pluginClass}DataAccess ${pluginFormat}DataAccess;
+
+  @Inject
+  public ${pluginClass}ProxyFacetImpl(final ${pluginClass}PathUtils ${pluginFormat}PathUtils,
+                             final ${pluginClass}DataAccess ${pluginFormat}DataAccess)
+  {
+    this.${pluginFormat}PathUtils = checkNotNull(${pluginFormat}PathUtils);
+    this.${pluginFormat}DataAccess = checkNotNull(${pluginFormat}DataAccess);
+  }
+
+  // HACK: Workaround for known CGLIB issue, forces an Import-Package for org.sonatype.nexus.repository.config
+  @Override
+  protected void doValidate(final Configuration configuration) throws Exception {
+    super.doValidate(configuration);
+  }
+
+  @Nullable
+  @Override
+  protected Content getCachedContent(final Context context) {
+    AssetKind assetKind = context.getAttributes().require(AssetKind.class);
+    TokenMatcher.State matcherState = ${pluginFormat}PathUtils.matcherState(context);
+    switch (assetKind) {
+      case PACKAGES:
+        return getAsset(${pluginFormat}PathUtils.buildAssetPath(matcherState, INDEX_HTML));
+      case ARCHIVE:
+        return getAsset(${pluginFormat}PathUtils.buildAssetPath(matcherState, CHANNELDATA_JSON));
+      default:
+        throw new IllegalStateException("Received an invalid AssetKind of type: " + assetKind.name());
+    }
+  }
+
+  @TransactionalTouchBlob
+  //protected Content getAsset(final String assetPath) {
```

Comment:
Can we remove this or should it go back to protected?

Reply:
I will go ahead and remove it. Wasn't sure about which way to go either.
Removed in this commit: b75a10b

This commit is a good example of how the "template" class changes, and then the related "reference" class should be updated to match that change. (The "Foo" format of the archetype IT).